.github/workflows/commit.yml

```diff
@@ -264,3 +222,38 @@ jobs:
           COSIGN_PASSWORD: ${{ secrets.COSIGN_PASSWORD }}
           TOOL: ${{ matrix.tool }}
           ENVIRONMENT: ${{ matrix.environment }}
+  finalize-the-release:
+    name: Finalize the release
+    needs: [bump-version, distribute]
+    if: "${{ github.event_name == 'push' && !startsWith(github.event.head_commit.message, 'Bump version: 2') }}"
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: "${{ needs.bump-version.outputs.git_tag }}"
+      - name: Download the SBOMs and Vuln scan results
+        uses: actions/download-artifact@v3
+        with:
+          path: ${{ runner.temp }}
+      - name: Publish the release to GitHub
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          name: ${{ steps.bump-version.outputs.git_tag }}
+          tag_name: ${{ steps.bump-version.outputs.git_tag }}
+          generate_release_notes: true
+          files: |
```

Comment:
This matches what I saw in experiments [here](https://github.com/SeisoLLC/easy_infra/actions/runs/5821731392/job/15784789848#step:13:200)